### ISSUE TITLE
[Airbrake] [Production] undefined method `for' for nil:NilClass

### ISSUE THREAD

**Issue by diraulo** [AgileVentures/WebsiteOne#2881]
**Airbrake error:** [#9826](https://herokuapp21498474herokucom.airbrake.io/projects/117119/groups/2357414467120659826)
**Airbrake project:** websiteone-production

**Error type:** `ActionView::Template::Error`
**Error message:** `` undefined method `for' for nil:NilClass ``
**Where:** `<no information>`
**Occurred at:** Nov 27, 2018 14:41:43 UTC
**First seen at:** Nov 27, 2018 14:41:43 UTC
**Occurrences:** 0 (0 since last deploy on `<no information>`)
**Severity:** `error`

**URL:** `<no information>`
**File:** `/PROJECT_ROOT/app/models/event_instance.rb`

**Backtrace:**
```
/PROJECT_ROOT/app/models/event_instance.rb:22:in for
/PROJECT_ROOT/app/presenters/base_presenter.rb:27:in method_missing
/PROJECT_ROOT/app/helpers/event_instances_helper.rb:8:in watchable?
/PROJECT_ROOT/app/views/event_instances/_index_basic_info.html.erb:14:in _app_views_event_instances__index_basic_info_html_erb___4537159940656139537_70194701593520
...
Caused by NoMethodError: undefined method `for' for nil:NilClass
```

**Comment by NickSchimek** [comment]
I am reopening this issue as it occurs multiple times a day and @mattwr18 discovered why it's happening and documented it in #websiteone-devops slack channel. Basically event_instances may have an event_id that is nil, and currently empty event_instances can be created by anyone.

To fix this I'd suggest the following action items be completed:

- [ ] Add test to capture non-logged in user creating an event_instance
- [ ] Add test to capture user creating an event_instance without all fields blank
- [ ] Add test to capture a valid event instance that doesn't belong to an event.
- [ ] Add validation to check an authorized user is making the request to update/create/edit the event_instance
- [ ] Add validations to check that all params are present as they should be if the user is using the UI
- [ ] Update https://github.com/AgileVentures/WebsiteOne/blob/develop/app/models/event_instance.rb#L22 to not raise an error if there is no event associated to the event_instance. This can be done by using the `&` operator.

**Comment by tansaku** [comment]
thanks @NickSchimek, great work @mattwr18 

I'd prioritise a quick deploy of:

>  Update https://github.com/AgileVentures/WebsiteOne/blob/develop/app/models/event_instance.rb#L22 to not raise an error if there is no event associated to the event_instance. This can be done by using the & operator.

if this is causing repeated problems

**Comment by NickSchimek** [comment]
I don't think it needs to be rushed, however one could argue it's a security issue.... hmmm I could take ownership of this issue and start/complete it around May 15th....... if no one else was interested in it.

**Comment by mattwr18** [comment]
I think @ciremoussadia was maybe going to have a look

**Comment by ciremoussadia** [comment]
@mattwr18 I'll be glad to pair with @NickSchimek on this. I think he will help me to better understand the context. So @NickSchimek let me know when u'll be free.

**Comment by mattwr18** [comment]
that's great! I'm also available to pair before the 15th, maybe we could just take care of the use case that @NickSchimek mentioned :point_up: that @tansaku referenced. It's my understanding that might fix the production failures.

We could put in a `PR` for that that could get merged right away, then we could focus on the other use cases, which @NickSchimek could help out with as well. What do you think @ciremoussadia?

**Comment by ciremoussadia** [comment]
@mattwr18 I'm ok with that.

**Comment by NickSchimek** [comment]
This will be partially fixed with #3171 

Issue #3143 is related to this. Updating #3143 issue to authenticate user as this issue will be closed with #3171
